core/src/qrcode/QRFormatInformation.h

```diff
@@ -24,19 +24,21 @@ namespace ZXing {
 namespace QRCode {
 
 /**
-* <p>Encapsulates a QR Code's format information, including the data mask used and
-* error correction level.</p>
-*
-* @author Sean Owen
-* @see DataMask
-* @see ErrorCorrectionLevel
-*/
+ * <p>Encapsulates a QR Code's format information, including the data mask used and
+ * error correction level.</p>
+ *
+ * @author Sean Owen
+ * @see DataMask
+ * @see ErrorCorrectionLevel
+ */
 class FormatInformation
 {
 public:
 	FormatInformation() = default;
 
-	static FormatInformation DecodeFormatInformation(uint32_t formatInfoBits1, uint32_t formatInfoBits2);
```

Comment:
I fail to understand why this "Factory" abstraction was introduced. It's late, so maybe that is the reason... ;) But why not simply add a `DecodeFormatInformation(uint32_t formatInfoBits)` overload that produces the MircoQRCode info?